src/model_theory/terms_and_formulas.lean

```diff
@@ -170,32 +211,120 @@ nonempty.intro (by constructor)
 variables (M)
 
 /-- A bounded formula can be evaluated as true or false by giving values to each free variable. -/
-@[simp] def realize_bounded_formula :
+def realize_bounded_formula :
```

Comment:
rename to `bounded_formula.realize`?

Reply:
I will try to also do this with terms and formulas, but I worry about the `bounded_formula` and `formula` projection notations applying simultaneously to formulas.